### DISCUSSION TITLE
Updating nerdctl on existing instances

### DISCUSSION THREAD

**Discussion by DennisRasey** [lima-vm/lima#551] [Q&A]
Nerdctl doesn't seem to get upgraded on an existing vm even though lima appears to download a newer version on startup.

`INFO[0000] Attempting to download the nerdctl archive from "https://github.com/containerd/nerdctl/releases/download/v0.16.0/nerdctl-full-0.16.0-linux-amd64.tar.gz"  digest="sha256:6dd955393fd781601a1cc7655d178fa539543f7a62f29e7c34a00e31ff4dddcf"`

```
lima@lima-debian-sid:~$ nerdctl --version
nerdctl version 0.15.0
```
 
Is this by design?
 
If not, I've added a small hack to `
/pkg/cidata/cidata.TEMPLATE.d/boot/40-install-containerd.sh
` that checks if we're running the latest version of nerdctl and allows the script to install/upgrade it if not.

**Comment by AkihiroSuda** [comment]
> Is this by design?

No, we should update nerdctl: https://github.com/lima-vm/lima/issues/303

I'm still not sure whether we should upgrade it by default or not.
RFC.

**Reply by DennisRasey** [reply]
We could add a shell script that allows the user to manually initiate/force an upgrade of nerdctl.

If that makes sense, I can move this to conversation to #303 and submit a PR for the change(s).

**Reply by jandubois** [reply]
> We could add a shell script that allows the user to manually initiate/force an upgrade of nerdctl.

I don't think it should be a shell script, but implemented by `limactl` itself, but am currently drawing a blank on suggesting a name.

`limactl start` should also display an INFO or WARN message when it detects that the nerdctl in cidata is newer than the installed version.

**Reply by DennisRasey** [reply]
> am currently drawing a blank on suggesting a name. 
> 

limactl update-nerdctl `<instance name> `?
